palette/colormap.go

```diff
@@ -0,0 +1,150 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package palette
+
+import (
+	"fmt"
+	"image/color"
+	"math"
+
+	"github.com/gonum/floats"
+)
+
+// FromPalette creates a ColorMap that maps values to colors based
+// on RGB interpolation among the colors in p.
+func FromPalette(p Palette) ColorMap {
```

Comment:
This name is not good. It will be called as `palette.FromPalette(p)` which is misleading: `NewColorMap`. It's also odd that you take a `Palette` and then immediately extract the colours. If that's what you want, make the signature `NewColorMap(c []color.Color) ColorMap`.